src/router/exchange_router.cairo

```diff
@@ -288,13 +288,13 @@ mod ExchangeRouter {
 
         fn create_deposit(ref self: ContractState, params: CreateDepositParams) -> felt252 {
             let data_store = self.data_store.read();
-            global_reentrancy_guard::non_reentrant_before(data_store);
+            // global_reentrancy_guard::non_reentrant_before(data_store);
```

Comment:
The `global_reentrancy_guard::non_reentrant_before` function is also called in the `create_deposit` function.
So I think there's one call too many, please, can you check if it's really this call that should be removed and not the one from the `create_deposit` function?